velox/exec/Unnest.cpp

```diff
@@ -192,13 +192,25 @@ RowVectorPtr Unnest::getOutput() {
       // Construct unnest column using Array elements wrapped using above
       // created dictionary.
       auto unnestBaseArray = currentDecoded.base()->as<ArrayVector>();
-      outputs[outputsIndex++] = identityMapping
-          ? unnestBaseArray->elements()
-          : wrapChild(
-                numElements,
-                elementIndices,
-                unnestBaseArray->elements(),
-                nulls);
+      auto elements = unnestBaseArray->elements();
+      auto type = elements->type();
+      if (type->isRow()) {
+        auto size = type->size();
+        auto row = elements->as<RowVector>();
```

Comment:
Do we need to handle the case when elements are dictionary encoded ?

Reply:
Thanks for the feedback, I believe `wrapChild` currently handles row elements with dictionary encoding:
```
outputs[outputsIndex++] = identityMapping
              ? row->childAt(j)
              : wrapChild(numElements, elementIndices, row->childAt(j), nulls);
```
Please correct me if I am wrong.

Reply:
@pramodsatya I'm trying to say that if 'element's is not a flat vector, then elements->as<RowVector>() will fail.

Reply:
Thanks for the clarification, when `LEGACY_UNNEST` is set to `false` Presto only unnests arrays of rows. Arrays of maps are not unnested into `keys, values` columns.

Reply:
Got it. I guess after this fix we would need to add a check in Prestissimo to fail fast if LEGACY_UNNEST == true.

Reply:
Thanks, quick question, could you please share why Prestissimo would fail when LEGACY_UNNEST == true? As previously suggested, I am looking at how the behavior of `Unnest` can be controlled based on the value of `LEGACY_UNNEST` property.

Reply:
I assume we would not want to carry forward legacy behaviors. These increase maintenance costs without adding value. Unless there is a need to support legacy unnest, I'd rather add a quick check and fail fast if legacy unnest is requested.

Reply:
Thanks @mbasmanova, it would be better to deprecate the legacy unnest. I will follow up with a PR on PrestoDb to fail fast when legacy unnest is set to true. Could you please help review/merge this PR? The documentation for `UnnestNode` is updated.